### PR TITLE
feat: implement risks for dataflow

### PR DESCRIPTION
**Pull request by vjerci**
## Description
This pr adds risks to dataflow. Risks are similar to dataflow's datatypes except that they are grouped by detector.
This pr also addresses and fixes a custom ruby detector bug. We were missing filenames and file language in source, since it was affecting dataflow (empty filenames) it was fixed as part of this pr

## Related
https://bearer.atlassian.net/jira/software/projects/AMA/boards/57?selectedIssue=AMA-3145

## Checklist

- [x] I've added test coverage that shows my fix or feature works as expected.
- [ ] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: pkg/detectors/sql/custom_detector/extract_arguments.go [hunk 1]
```diff
@@ -6,9 +6,10 @@ import (
 	"github.com/bearer/curio/pkg/parser/nodeid"
 	"github.com/bearer/curio/pkg/report/schema"
 	schemadatatype "github.com/bearer/curio/pkg/report/schema/datatype"
+	"github.com/bearer/curio/pkg/util/file"
 )
 
-func (detector *Detector) ExtractArguments(node *parser.Node, idGenerator nodeid.Generator) (map[parser.NodeID]*schemadatatype.DataType, error) {
+func (detector *Detector) ExtractArguments(node *parser.Node, idGenerator nodeid.Generator, filinfo *file.FileInfo, filepath *file.Path) (map[parser.NodeID]*schemadatatype.DataType, error) {
```
Comment (cfabianski) [inline pkg/detectors/sql/custom_detector/extract_arguments.go]:
```suggestion
func (detector *Detector) ExtractArguments(node *parser.Node, idGenerator nodeid.Generator, fileinfo *file.FileInfo, filepath *file.Path) (map[parser.NodeID]*schemadatatype.DataType, error) {
```

File: pkg/report/output/dataflow_datatype_test.go [hunk 2]
```diff
@@ -39,17 +39,15 @@ func TestDataflow(t *testing.T) {
...
-								Locations: []output.Location{
-									{Filename: "./users.rb", LineNumber: 25},
-								},
+			Want: []types.Datatype{
+				{
+					Name: "User",
+					Detectors: []types.DatatypeDetector{
+						{
+							Name:   "ruby",
+							Stored: true,
```
Comment (cfabianski) [inline pkg/report/output/dataflow_datatype_test.go:48]: How do you know it's stored here?

File: pkg/report/output/dataflow/dataflow.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			if (detectionType) == allowedDetection {
+				isDataflow = true
+			}
+		}
+
+		if !isDataflow {
+			continue
+		}
+
+		var castedDetection detections.Detection
```
Comment (elsapet) [inline pkg/report/output/dataflow/dataflow.go]: Suggestion to change the variable name: `cast` reads more naturally to me (unless `casted` is a Go convention, in which case ignore me :) )

```suggestion
		var castDetection detections.Detection
```

**Review by elsapet — DISMISSED** [review]

File: pkg/report/output/dataflow_datatype_test.go (hunk 2 above)
Comment (vjerci) [reply pkg/report/output/dataflow_datatype_test.go:48]: will handle it in separate pr

**Review by cfabianski — APPROVED** [review]

**Pull request merged**